python/lancedb/remote/client.py

```diff
@@ -170,3 +178,50 @@ def query(self, table_name: str, query: VectorQuery) -> VectorQueryResult:
         """Query a table."""
         tbl = self.post(f"/v1/table/{table_name}/query/", query, deserialize=_read_ipc)
         return VectorQueryResult(tbl)
+    
+    def mount_retry_adapter_for_table(self, table_name: str) -> None:
+        """
+        Adds an http adapter to session that will retry retryable requests to the table.
+        """
+        retry_options = retry_adapter_options(methods=["GET", "POST"])
+        retry_adapter_instance = retry_adapter(retry_options)
+        session = self.session
+
+        session.mount(urljoin(self.url, f"/v1/table/{table_name}/query/"), retry_adapter_instance)
+        session.mount(urljoin(self.url, f"/v1/table/{table_name}/describe/"), retry_adapter_instance)
+        session.mount(urljoin(self.url, f"/v1/table/{table_name}/index/list/"), retry_adapter_instance)
+
+def retry_adapter_options(methods = ["GET"]) -> Dict[str, Any]:
+    return {
+        "retries": int(os.environ.get("LANCE_CLIENT_MAX_RETRIES", "3")),
+        "connect_retries": int(os.environ.get("LANCE_CLIENT_CONNECT_RETRIES", "3")),
+        "read_retries": int(os.environ.get("LANCE_CLIENT_READ_RETRIES", "3")),
+        "backoff_factor": float(os.environ.get("LANCE_CLIENT_RETRY_BACKOFF_FACTOR", "0.25")),
+        "statuses": [
+            int(i.strip()) for i in os.environ.get(
+                "LANCE_CLIENT_RETRY_STATUSES", "500, 502, 503"
+            ).split(",")
+        ],
+        "methods": methods,
+    }
+
+def retry_adapter(options: Dict[str, Any]) -> HTTPAdapter:
+    total_retries = options["retries"]
+    connect_retries = options["connect_retries"]
+    read_retries = options["read_retries"]
+    backoff_factor = options["backoff_factor"]
+    statuses = options["statuses"]
+    methods = frozenset(options["methods"])
+    logging.debug(f"Setting up retry adapter with {total_retries} retries," + 
+                  f"connect retries {connect_retries}, read retries {read_retries}," +
+                  f"backoff factor {backoff_factor}, statuses {statuses}, " +
+                  f"methods {methods}")
+
+    return HTTPAdapter(max_retries=Retry(
```

Comment:
is this exponential retry? Also, anyway we can add jitter to the back off?

Reply:
It is exponential:
```
sleep = backoff_factor * (2 ** retry_num)
```
So by default (backoff_factor = 0.25 * total retries = 3), we sleep for 0.5, 1, and 2 seconds

I will add jitter as well